aquadoggo/src/replication/errors.rs

```diff
@@ -49,3 +52,28 @@ pub enum TargetSetError {
     #[error("Target set contains unsorted or duplicate schema ids")]
     UnsortedSchemaIds,
 }
+
+#[derive(Error, Debug, PartialEq)]
+pub enum DuplicateSessionRequestError {
+    #[error("Remote sent two sync requests for session with id {0}")]
+    InboundPendingSession(u64),
+
+    #[error("Tried to initialise duplicate inbound replication for already established session with id {0}")]
+    InboundEstablishedSession(u64),
+
+    #[error("Tried to initialise duplicate inbound replication for completed session with id {0}")]
+    InboundDoneSession(u64),
+
+    #[error(
+        "Tried to initialise duplicate inbound replication session for existing target set {0:?}"
+    )]
+    InboundExistingTargetSet(TargetSet),
+
+    #[error(
+        "Tried to initialise duplicate outbound replication session for existing target set {0:?}"
+    )]
+    OutboundExistingTargetSet(TargetSet),
+
+    #[error("Tried to initialise duplicate outbound replication session with id {0}")]
+    Outbound(u64),
```

Comment:
A load of errors relating to our de-duplication logic in sync manager..... Not super happy with this yet, but it's nice to have custom error types for these cases.

Reply:
I like it, that might make it easier to debug in case something goes wrong